atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/thrift/ThriftObjectSizeUtils.java

```diff
@@ -284,14 +286,26 @@ private static <T> long getBytesSize(T data, Function<T, ByteBuffer> toByteBuffe
         return byteBuffer.remaining();
     }
 
-    public static <T> long getCollectionSize(Collection<T> collection, Function<T, Long> sizeFunction) {
+    public static <T> long getCollectionSize(Collection<T> collection, ToLongFunction<T> sizeFunction) {
         if (collection == null) {
             return getNullSize();
         }
 
-        long sum = 0;
-        for (T item : collection) {
-            sum += sizeFunction.apply(item);
+        long sum = 0L;
+        if (collection instanceof List && collection instanceof RandomAccess) {
```

Comment:
Also, how would you feel if we made private method for these two? I wouldn't imagine it would add too much overhead, and may make things slightly more readable?

so (psuedocode): 

```
if (isListAndRandomAccess(collection)) {
    return iterateByIndexAccess(...)
}
return iterateByIterator(...)
```